docs/src/examples/components/Popup/Variations/PopupExamplePosition.shorthand.tsx

```diff
@@ -1,54 +1,91 @@
-  const { position, align, icon, padding } = props
+const PopupExamplePosition = () => {
+  const [open] = useBooleanKnob({ name: 'open shorthand', initialValue: true })
 
-  return (
-    <Popup
-      align={align}
-      position={position}
-      trigger={<Button icon={icon} styles={{ padding, height: '38px', minWidth: '64px' }} />}
-      content={{
-        content: (
-          <p>
-            The popup is rendered {position} the trigger
-            <br />
-            aligned to the {align}.
-          </p>
-        ),
-      }}
-    />
+  const [position] = useSelectKnob<Position>({
+    name: 'position shorthand',
+    initialValue: 'above',
+    values: ['above', 'below', 'before', 'after'],
+  })
+
+  const [positionBeforeOrAfter, setPositionBeforeOrAfter] = React.useState<boolean>(
+    isPositionBeforeOrAfter(position),
   )
-}
 
-const triggers = [
-  { position: 'above', align: 'start', icon: 'arrow circle up', padding: '5px 42px 18px 5px' },
-  { position: 'above', align: 'center', icon: 'arrow circle up', padding: '5px 5px 18px 5px' },
-  { position: 'above', align: 'end', icon: 'arrow circle up', padding: '5px 5px 18px 42px' },
-  { position: 'below', align: 'start', icon: 'arrow circle down', padding: '18px 42px 5px 5px' },
-  { position: 'below', align: 'center', icon: 'arrow circle down', padding: '18px 5px 5px 5px' },
-  { position: 'below', align: 'end', icon: 'arrow circle down', padding: '18px 5px 5px 42px' },
-  { position: 'before', align: 'top', icon: 'arrow circle left', padding: '5px 42px 18px 5px' },
-  { position: 'before', align: 'center', icon: 'arrow circle left', padding: '5px 42px 5px 5px' },
-  { position: 'before', align: 'bottom', icon: 'arrow circle left', padding: '18px 42px 5px 5px' },
-  { position: 'after', align: 'top', icon: 'arrow circle right', padding: '5px 5px 18px 42px' },
-  { position: 'after', align: 'center', icon: 'arrow circle right', padding: '5px 5px 5px 42px' },
-  { position: 'after', align: 'bottom', icon: 'arrow circle right', padding: '18px 5px 5px 42px' },
-]
-
-const PopupExamplePosition = () => (
-  <Grid columns="repeat(3, 30px)" variables={{ padding: '30px', gridGap: '80px' }}>
-    {triggers.map(({ position, align, icon, padding }) => (
-      <PopupWithButton
-        position={position}
+  const [align] = useSelectKnob<Alignment>(
+    positionBeforeOrAfter
+      ? {
+          name: 'h-align shorthand',
+          initialValue: 'top',
+          values: ['top', 'center', 'bottom'],
+        }
+      : {
+          name: 'v-align shorthand',
+          initialValue: 'start',
+          values: ['start', 'center', 'end'],
+        },
+  )
+
+  React.useEffect(() => setPositionBeforeOrAfter(isPositionBeforeOrAfter(position)), [position])
+
+  const buttonStyles = { padding: paddings[position][align], height: '38px', minWidth: '64px' }
+
+  return (
+    <Grid columns="1" variables={{ padding: '100px 0' }} styles={{ justifyItems: 'center' }}>
```

Comment:
Is `Grid` the best component for 0D layout?

Reply:
I don't know :( I used it because it was used in https://github.com/stardust-ui/react/blob/5c2688a9715b3b5e3925cc8da59420eada40a5c7/docs/src/examples/components/Popup/Variations/PopupExampleOffset.shorthand.tsx#L15